packages/blade/src/components/TopNav/_decisions/decisions.md

```diff
@@ -122,6 +146,155 @@ The container for the right-aligned actions like search, buttons, and user profi
 
 ### TabNav
 
+The horizontal navigation bar that contains multiple navigation items, each represented by TabNavItem. 
+
+TabNav's interaction is as follows: 
+
+1. There might already be items under More even at the highest screen size. These items might need a custom order.
+2. Once the screen size starts getting reduced, more items will start moving into More.
+3. The order of these items would follow a logical order. They keep getting added ahead of the already existing items. So the last item to go into More is the first item that you see when you click on More.
+
+TabNav automatically handles all these edge cases & responsiveness and let's you focus on the rendering of the items via the [render props](https://reactpatterns.js.org/docs/function-as-child-component/) pattern.
+
+```jsx
+<TabNav
+  items={[
+    { href: '/home', title: 'Home' },
+    { href: '/payroll', title: 'Payroll' },
+    { href: '/payments', title: 'Payments' },
+    {
+      href: '/rize',
+      title: 'rize',
+      description: 'rize description',
+      icon: RizeIcon,
+      // force this item to always be inside "more" regardless of screen size
+      isAlwaysOverflowing: true,
+    },
+  ]}
+>
+  {({ items, overflowingItems }) => {
+    return (
+      <>
+        <TabNavItems>
```

Comment:
It holds all the horizontal items and by it's width we also determine the overflowing behaviour. 

<img width="522" alt="image" src="https://github.com/user-attachments/assets/544e0f8b-0ff9-49d6-9aba-cc6dc6d43577">

Reply:
Makes sense, was wondering why we added an additional component since it wasn't present before right? 
Will this be a blade component alongside TabNavItem which will be exposed to the consumer?

Reply:
As i said 

> it's width we also determine the overflowing behaviour.

This is the container that holds your items and when it's width shrinks we automatically detect it via the ResizeObserver.